### PR TITLE
Push app context in ApiClient

### PR DESCRIPTION
**Pull request by thomasst**
Execute ApiClient calls within a separate app context.

### CODE REVIEW

**Comment by wojcikstefan** [comment]
Does `local_request` not have the same problem because `test_request_context` creates a new app context?

File: flask_common/client.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def open(self, *args, **kwargs):
         api_key = kwargs.pop('api_key', self.api_key)
         if 'headers' not in kwargs:
             kwargs['headers'] = self.get_headers(api_key)
-        return super(ApiClient, self).open(*args, **kwargs)
+        with current_app.app_context():
+            return super(ApiClient, self).open(*args, **kwargs)
```
Comment (wojcikstefan) [inline flask_common/client.py]: Does it ever make sense to reuse the same app context? If not, should we move it to `Client.open`?

Comment (thomasst) [inline flask_common/client.py]: The problem with `Client.open` is that it then also applies to a session client, where you potentially want to keep your session for tests. Several tests fail if you have it in `Client`.

**Comment by thomasst** [comment]
> Does local_request not have the same problem because test_request_context creates a new app context?

`local_request` doesn't preprocess the request, i.e. before/after-request hooks are not called, so no CSRF validation takes place.

**Comment by wojcikstefan** [comment]
> `local_request` doesn't preprocess the request, i.e. before/after-request hooks are not called, so no CSRF validation takes place.

This should probably be documented somewhere as it might be dangerous in some scenarios.

**Pull request merged**